src/code/z_camera.c

```diff
@@ -543,323 +868,7415 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+    viewForwardsUnitX = Math_SinS(viewYaw);
+    viewForwardsUnitZ = Math_CosS(viewYaw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD2F0.s")
+    focalActorHeight = Camera_GetFocalActorHeight(camera);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD3E4.s")
+    checkOffsetY = focalActorHeight * 1.2f;
+    nearDist = focalActorHeight * 1.0f;
+    farDist = focalActorHeight * 2.5f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD44C.s")
+    focalActorPos.x = camera->focalActorPosRot.pos.x;
+    focalActorPos.y = camera->focalActorFloorHeight + checkOffsetY;
+    focalActorPos.z = camera->focalActorPosRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD634.s")
+    nearPos.x = focalActorPos.x + (nearDist * viewForwardsUnitX);
+    nearPos.y = focalActorPos.y;
+    nearPos.z = focalActorPos.z + (nearDist * viewForwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD6CC.s")
+    if (shouldInit || ((camera->play->state.frames % 2) == 0)) {
+        sFarColChk.pos.x = focalActorPos.x + (farDist * viewForwardsUnitX);
+        sFarColChk.pos.y = focalActorPos.y;
+        sFarColChk.pos.z = focalActorPos.z + (farDist * viewForwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD6F8.s")
+        Camera_BgCheckInfo(camera, &focalActorPos, &sFarColChk);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CD834.s")
+        if (shouldInit) {
+            sFloorYNear = sFloorYFar = camera->focalActorFloorHeight;
+        }
+    } else {
+        farDist = OLib_Vec3fDistXZ(&focalActorPos, &sFarColChk.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CDA14.s")
+        sFarColChk.pos.x += sFarColChk.norm.x * 5.0f;
+        sFarColChk.pos.y += sFarColChk.norm.y * 5.0f;
+        sFarColChk.pos.z += sFarColChk.norm.z * 5.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CDB6C.s")
+        if (nearDist > farDist) {
+            nearDist = farDist;
+            sFloorYNear = sFloorYFar = Camera_GetFloorYLayer(camera, &floorNorm, &sFarColChk.pos, &bgId);
+        } else {
+            sFloorYNear = Camera_GetFloorYLayer(camera, &floorNorm, &nearPos, &bgId);
+            sFloorYFar = Camera_GetFloorYLayer(camera, &floorNorm, &sFarColChk.pos, &bgId);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CDE6C.s")
+        if (sFloorYNear == BGCHECK_Y_MIN) {
+            sFloorYNear = camera->focalActorFloorHeight;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE2B8.s")
+        if (sFloorYFar == BGCHECK_Y_MIN) {
+            sFloorYFar = sFloorYNear;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE5E0.s")
+    floorYDiffNear = (sFloorYNear - camera->focalActorFloorHeight) * 0.8f;
+    floorYDiffFar = (sFloorYFar - camera->focalActorFloorHeight) * 0.19999999f;
```

Comment:
Either `(20.0f * 0.01f)` or `(0.02f * 10.0f)` seem to work not sure if either has any larger significance in camera. 
Note `0.8f` can't match either of this styles which is sad :( .

Reply:
Bump